### PR TITLE
Make users send and accept friend requests

### PR DESCRIPTION
**Pull request by luederb**

### CODE REVIEW

File: backend/src/main/java/org/example/backend/model/FriendRequest.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class FriendRequest {
+    private String id;
+    private User sender;
+    private User receiver;
+    private String status;
```
Comment (MaHambach) [inline backend/src/main/java/org/example/backend/model/FriendRequest.java]: Maybe use an Enum for the status? What are the possible states? OPEN, ACCEPTED, DECLINED?

File: backend/src/test/java/org/example/backend/model/FriendRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package org.example.backend.model;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+class FriendRequestTest {
```
Comment (MaHambach) [inline backend/src/test/java/org/example/backend/model/FriendRequestTest.java]: You shouldn't need to test those, if you add a lombok.config to your backend/src folder.

Comment (MaHambach) [reply backend/src/test/java/org/example/backend/model/FriendRequestTest.java]: those = your models.

File: backend/src/test/java/org/example/backend/service/UserServiceTest.java [hunk 3]
```diff
@@ -137,4 +163,50 @@ void getAllUsersTest() {
...
+    @Test
+    void sendFriendRequestTest() {
+        when(userRepository.save(any(User.class))).thenReturn(user1, user2);
+        when(friendRequestRepository.save(any(FriendRequest.class))).thenReturn(friendRequest);
+
+        FriendRequest result = userService.sendFriendRequest(user1, user2);
+
+        assertEquals(user1.getId(), result.getSender().getId());
+        assertEquals(user2.getId(), result.getReceiver().getId());
+        assertEquals(UserService.STATUS_PENDING, result.getStatus());
```
Comment (MaHambach) [inline backend/src/test/java/org/example/backend/service/UserServiceTest.java]: Maybe add `assertEquals(friendRequest.getId(), result.getId());`.

File: frontend/src/Types/Friend.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export type Friend = {
+    id: string;
+    name: string;
+    firstName: string;
+    lastName: string;
+    email: string;
+    phone?: number;
+    bio?: string;
+    picture: string;
```
Comment (MaHambach) [inline frontend/src/Types/Friend.ts:9]: Does every user have a picture´?

File: frontend/src/components/Network/FriendRequests/FriendRequests.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                        <li key={friendRequest.id} className="friend-request">
+                            <p>{friendRequest.sender.name}</p>
+                            <button className="classic-button" onClick={() => acceptFriendRequest(friendRequest.receiver.id, friendRequest.id)}>accept </button>
+                        </li>
+                    ))}
+                </ul>
+            }
+        </div>
+    )
+}
```
Comment (MaHambach) [inline frontend/src/components/Network/FriendRequests/FriendRequests.tsx:57]: I like the way you encapsulate the friend-request management here. I will think about copying this way of doing things. ;)

**Review by MaHambach — CHANGES_REQUESTED** [review]
Got some questions. ^^ Please ping me in slack when you have answered.

File: backend/src/main/java/org/example/backend/model/FriendRequest.java (hunk 1 above)
Comment (luederb) [reply backend/src/main/java/org/example/backend/model/FriendRequest.java]: Thats a good idea. I changed the String status do a enum like this: 

```
package org.example.backend.model;

import lombok.Getter;

@Getter
public enum Status {
    PENDING("pending"),
    ACCEPTED("accepted"),
    REJECTED("rejected");

    private final String statusValue;

    Status(String statusValue) {
        this.statusValue = statusValue;
    }

}
```

File: backend/src/test/java/org/example/backend/model/FriendRequestTest.java (hunk 2 above)
Comment (luederb) [reply backend/src/test/java/org/example/backend/model/FriendRequestTest.java]: oh, good to know 🥳
I was already trying to find a solution, to not manually test them. Didn't really think, that everytime I use the lombok methods, I also have to test them 🤔
Thanks for the tip 😊

File: frontend/src/Types/Friend.ts (hunk 4 above)
Comment (luederb) [reply frontend/src/Types/Friend.ts:9]: Yes, I always get a picture from the google api. Either it's the picture, that the user uploaded, or its just a placeholder picture with the first letter of the name of the user

File: backend/src/test/java/org/example/backend/service/UserServiceTest.java (hunk 3 above)
Comment (luederb) [reply backend/src/test/java/org/example/backend/service/UserServiceTest.java]: Good idea. Thanks :-)

File: frontend/src/components/Network/FriendRequests/FriendRequests.tsx (hunk 5 above)
Comment (luederb) [reply frontend/src/components/Network/FriendRequests/FriendRequests.tsx:57]: Thanks 😊

**Review by MaHambach — APPROVED** [review]
LGTM ^^

**Review by MaHambach — APPROVED** [review]
Test

**Review by MaHambach — APPROVED** [review]

**Pull request merged**